core/modules/node/node.types.inc

```diff
@@ -287,12 +287,27 @@ function node_type_form($form, &$form_state, $type = NULL) {
   );
 
   $form['actions'] = array('#type' => 'actions');
+  // If this is a new content type and the Field UI module is enabled, add a
+  // 'Save and add fields' button.
+  if ($type->module === 'node' && empty($type->type) && module_exists('field_ui')) {
+    $form['actions']['add_fields'] = array(
```

Comment:
what I'm worried about here is that the name of the form element is being changed.   That means that if contrib or an end user is form_alter'ing this form and trying to referece this element this will break their site.